src/smt.rs

```diff
@@ -186,32 +186,39 @@ fn empty_hash() -> Vec<u8> {
 }
 
 fn tree_hasher(node_hashes: &[Vec<u8>], structure: &[u8], height: Height) -> Vec<u8> {
-    if node_hashes.len() == 1 {
-        return node_hashes[0].clone();
-    }
-    let mut next_hashes = vec![];
-    let mut next_structure = vec![];
-    let mut i = 0;
-
-    while i < node_hashes.len() {
-        if structure[i] == height.into() {
-            let branch = [node_hashes[i].clone(), node_hashes[i + 1].clone()].concat();
-            let hash = branch_hash(&branch);
-            next_hashes.push(hash);
-            next_structure.push(structure[i] - 1);
+    let mut node_hashes: Vec<Vec<u8>> = node_hashes.to_vec();
```

Comment:
strong type `NestedVec` instead of `Vec<Vec<u8>>`

Reply:
Done.